src/som/interpreter/actors/Actor.java

```diff
@@ -160,12 +226,39 @@ public synchronized void send(final EventualMessage msg) {
 
     logMessageAddedToMailbox(msg);
 
-    if (!isExecuting) {
+    // actor remains dormant until the expected message arrives
+    if ((!isExecuting) && replayCanProcess(msg)) {
```

Comment:
could we solve this better by overriding `send()` in the `TracingActor`?
I'd like if all this trace and replay code would be in that class.